src/new_index/mempool.rs

```diff
@@ -577,6 +503,68 @@ impl Mempool {
             .get(asset_id)
             .map_or_else(|| vec![], |entries| self._history(entries, limit))
     }
+
+    pub fn download_new_mempool_txs(
+        daemon: &Daemon,
+        old_txids: &HashSet<Txid>,
+        new_txids: &HashSet<Txid>
+    ) -> Result<Vec<Transaction>>{
+        let t = Instant::now();
+
+        let txids: Vec<&Txid> = (*new_txids).difference(old_txids).collect();
+        let transactions = match daemon.gettransactions(&txids) {
+            Ok(txs) => txs,
+            Err(err) => {
+                warn!("failed to get {} transactions: {}", txids.len(), err); // e.g. new block or RBF
```

Comment:
Errors here already bubble to `update()` which prints a `warn!()`, so this would result in duplicated warning messages for the same error.

I suggest changing the warning message in `update()` to include the caught error, then simplify the error handling code here to `let transactions = daemon.gettransactions(&txids).chain_err(|| format!("failed to get {} transactions", txids.len()))?;`.